dwave/cloud/config/models.py

```diff
@@ -0,0 +1,303 @@
+    logger.debug("parsed headers: %r", headers_dict)
+    config['headers'] = headers_dict
+
+    # parse optional client certificate
+    # (if string, path to ssl client cert file (.pem). If tuple, (‘cert’, ‘key’) pair.)
+    client_cert = config.get('client_cert')
+    client_cert_key = config.get('client_cert_key')
+    if client_cert_key is not None:
+        if client_cert is not None:
+            client_cert = (client_cert, client_cert_key)
+        else:
+            raise ValueError(
+                "Client certificate key given, but the cert is missing")
+    logger.debug("parsed client cert: %r", client_cert)
+    config['cert'] = client_cert
+
+    # parse solver
+    solver = config.get('solver')
+    if isinstance(solver, str):
+        solver = solver.strip()
+    if not solver:
+        solver_def = {}
+    elif isinstance(solver, abc.Mapping):
+        solver_def = solver
+    elif isinstance(solver, str):
+        # support features dict encoded as JSON in our config INI file
+        try:
+            solver_def = json.loads(solver)
+        except Exception:
+            # unparseable (or non-dict) json, assume string name for solver
+            # we'll deprecate this eventually, but for now just convert it to
+            # features dict (equality constraint on full solver name)
+            logger.debug("Invalid solver JSON, assuming string name: %r", solver)
+            solver_def = dict(name__eq=solver)
+        else:
+            # if valid json, has to be a dict
+            if not isinstance(solver_def, abc.Mapping):
+                logger.debug("Non-dict solver JSON, assuming string name: %r", solver)
+                solver_def = dict(name__eq=solver)
+    else:
+        raise ValueError("Expecting a features dictionary or a string name for 'solver'")
+    logger.debug("parsed solver definition: %r", solver_def)
+    config['solver'] = solver_def
+
+    # polling
+    prefix = 'poll_'
+    config['polling_schedule'] = {k[len(prefix):]: v for k, v in config.items()
+                                  if k.startswith(prefix)}
+
+    # retry
+    prefix = 'http_retry_'
+    config['request_retry'] = {k[len(prefix):]: v for k, v in config.items()
+                               if k.startswith(prefix)}
+
+    return ClientConfig.model_validate(config)
+
+
+def dump_config_v1(config: ClientConfig) -> dict:
+    """Dump :class:`ClientConfig` to a flat dict of v1 config options.
+
+    Note: consider this to be an approximate inverse of :func:`validate_config_v1`::
+
+        dump_config_v1(validate_config_v1(raw_config)) ~= raw_config
+
+    The inverse might not be exact due to field type casting during validation.
+    E.g. original boolean value of 'off' or '0' will be dumped as 'false'.
+    """
+
+    raw_config = config.model_dump()
+
+    # invert transformations during validation
+    if h := raw_config['headers']:
+        raw_config['headers'] = '\n'.join("{k}: {v}" for k, v in h.items())
```

Comment:
I assume this should be the values of `k` and `v` (using an f-string)?
```suggestion
        raw_config['headers'] = '\n'.join(f"{k}: {v}" for k, v in h.items())
```

Reply:
Great catch, thanks.